rust/ql/test/library-tests/type-inference/main.rs

```diff
@@ -24,36 +24,36 @@ mod field_access {
 
     fn simple_field_access() {
         let x = MyThing { a: S };
-        println!("{:?}", x.a);
+        println!("{:?}", x.a); // $ field=MyThing
     }
 
     fn generic_field_access() {
         // Explicit type argument
-        let x = GenericThing::<S> { a: S };
-        println!("{:?}", x.a);
+        let x = GenericThing::<S> { a: S }; // $ type=x:A.S
```

Comment:
What does this annotation mean exactly?

Reply:
This means that `x` has a type which at parameter position `A` contains `S`. In this case `x` has the type `GenericThing<S>` and `GenericThing` has a type parameter called `A`, hence `x` has `S` at `A`. For this line we could also add `$ type=x:GenericThing` since the "root" of `x`'s type is `GenericThing`.